src/query_ctrl.ts

```diff
@@ -146,12 +146,26 @@ export class HeroicQueryCtrl extends QueryCtrl {
         break;
       }
       case "action": {
-        this.queryModel.removeSelectPart(selectParts, part);
-        this.refresh();
+        if (evt.action.value === "remove-part") {
+          this.queryModel.removeSelectPart(selectParts, part);
+          this.refresh();
+        } else {
+          const category = _.find(this.selectMenu, menu => menu.text === evt.action.value);
```

Comment:
this is logic for converting an aggregation from one type (collapse, foreach, groupby) to another type